### PR TITLE
Action get_github_release authentication for private repos

### PR DESCRIPTION
**Pull request by czechboy0**
- added `User-Agent` header
- when `api_token` is passed in, also adds an `Authorization` header (useful for private repositories)

### CODE REVIEW

File: lib/fastlane/actions/get_github_release.rb [hunk 1]
```diff
@@ -8,7 +8,14 @@ class GetGithubReleaseAction < Action
       def self.run(params)
         Helper.log.info "Verifying release on GitHub (#{params[:url]}: #{params[:version]})"
         require 'excon'
-        result = JSON.parse(Excon.get("https://api.github.com/repos/#{params[:url]}/releases").body)
+        require 'base64'
+        headers = { 'User-Agent' => 'fastlane-get_github_release' }
```
Comment (KrauseFx) [inline lib/fastlane/actions/get_github_release.rb]: Do we need a user-agent?

Comment (czechboy0) [inline lib/fastlane/actions/get_github_release.rb]: Yes: https://developer.github.com/v3/#user-agent-required

**Comment by KrauseFx** [comment]
Thanks for your contribution, this looks great. It does however change the behaviour when the tag is not available or some error occurs. Up until now, the action would `return nil` and show a warning. The newly added code raises an exception if something goes wrong.

**Comment by czechboy0** [comment]
Ah of course. So which behavior would you prefer? Downgrading those messages from exceptions to error logs but letting the action finish?

**Comment by KrauseFx** [comment]
Yes, this should just be `Helper.logger.error "text"` :+1:

**Comment by czechboy0** [comment]
Ok. How about yours `raise "No release notes found for #{params[:version]}"`, should I change that as well?

**Comment by KrauseFx** [comment]
```ruby
             if current['body'].to_s.length > 0
               Actions.lane_context[SharedValues::GET_GITHUB_RELEASE_INFO] = current
               Helper.log.info "Version is already live on GitHub.com 🚁"
               return current
             else
               raise "No release notes found for #{params[:version]}"
             end
```

Oh, that's wrong anyway, as it requires a body to be set. I'll update that part. Thanks.

**Comment by KrauseFx** [comment]
Thanks for the pull request @czechboy0 :+1:

**Pull request merged**